ZipBuilder/Sources/oss-manifest-generator/main.swift

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+import ArgumentParser
+import ManifestReader
+
+struct OSSManifestGenerator: ParsableCommand {
+  /// The root of the Firebase git repo.
+  @Option(help: "The root of the firebase-ios-sdk checked out git repo.",
+          transform: URL.init(fileURLWithPath:))
+  var gitRoot: URL
+
+  /// A file URL to a textproto with the contents of a `FirebasePod_Release` object. Used to verify
+  /// expected version numbers.
+  @Option(name: .customLong("releasing-pods"),
+          help: "The file path to a textproto file containing all the releasing Pods, of type `ZipBuilder_Release`.",
+          transform: URL.init(fileURLWithPath:))
+  var currentRelease: URL
+
+  @Option(help: "The file path to a textproto file containing all existing Pods, of type `ZipBuilder_FirebaseSDKs.",
+          transform: URL.init(fileURLWithPath:))
+  var existingVersions: URL
+
+  mutating func validate() throws {
+    guard FileManager.default.fileExists(atPath: gitRoot.path) else {
+      throw ValidationError("git-root does not exist: \(gitRoot.path)")
+    }
+
+    guard FileManager.default.fileExists(atPath: currentRelease.path) else {
```

Comment:
Optional idea - replace both of these with a g3 repo root - with default being head

Reply:
I do like that idea although then we'll have to use the google3 relative path in the code and I think that's not preferred.